### DISCUSSION TITLE
Hide navigation bar?

### DISCUSSION THREAD

**Discussion by incw** [adrielcafe/voyager#322] [General]
What solution should I use to hide the bottom navigation on a specific screen?

**Comment by Mohamed-Mabrouki** [comment]
hello the solution is simple 
here is the code i added to voyager tab navigation sample to change the visibility depending on the current screen: https://github.com/Mohamed-Mab/voyager-tabNavigation-visibility

**Reply by sawirricardo** [reply]
Thank you @Mohamed-Mab, another question....

I have the detailScreen, which has a custom back button, when if clicked, will popUntilRoot(), AND, change the current tab to the other tab (which also depends on that onClick function).... How can we do that?

**Reply by techjd** [reply]
Hello @Mohamed-Mab , Your solution crashes on Android , when the app comes to foreground from recents . As we are passing onClick lambda as a parameter it's not able to parcelize that . Any solutions for this ?

**Reply by Mohamed-Mabrouki** [reply]
@techjd  you can pass @Trancient annotation to onClick lambda parameter and it should work fine

**Reply by Ayush-Kotlin-Dev** [reply]
Thanks @Mohamed-Mabrouki  , thanks for the solution.

**Comment by siftoshka** [comment]
I have found another solution for this case. 
So the structure is like this for example: StartScreen -> (HomeTab, SettingsTab)
The StartScreen has its own ViewModel. I am injecting this viewModel for StartScreen and tab screens(HomeScreen and SettingsScreen). 

Whenever I am in HomeScreen, I send an event to StartViewModel to change uiState (isNavBarVisible to false), and on StartScreen just observe uiState and wrap BottonNavBar in if state.

In this case, I am not using callbacks and it is cleaner for me. 

I can add code example if needed

**Reply by techjd** [reply]
Hello @siftoshka , Can you provide a sample code implementation ?
